integration-tests/src/test_loop/tests/contract_distribution.rs

```diff
@@ -0,0 +1,187 @@
+use itertools::Itertools;
+use near_async::test_loop::TestLoopV2;
+use near_async::time::Duration;
+use near_chain_configs::test_genesis::TestGenesisBuilder;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::types::{AccountId, BlockHeight};
+use rand::Rng;
+
+use crate::test_loop::builder::TestLoopBuilder;
+use crate::test_loop::env::{TestData, TestLoopEnv};
+use crate::test_loop::utils::transactions::{
+    call_contract, check_txs, deploy_contract, make_accounts,
+};
+use crate::test_loop::utils::ONE_NEAR;
+
+const NUM_ACCOUNTS: usize = 40;
+const EPOCH_LENGTH: u64 = 10;
+const GENESIS_HEIGHT: u64 = 1000;
+
+const NUM_BLOCK_AND_CHUNK_PRODUCERS: usize = 8;
+const NUM_CHUNK_VALIDATORS_ONLY: usize = 16;
+const NUM_RPC: usize = 1;
+const NUM_VALIDATORS: usize = NUM_BLOCK_AND_CHUNK_PRODUCERS + NUM_CHUNK_VALIDATORS_ONLY;
+
+/// Tests a scenario that different contracts are deployed to a number of accounts and
+/// these contracts are called from a set of accounts.
+#[test]
+fn test_contract_distribution_testloop() {
+    init_test_logger();
+    let accounts = make_accounts(NUM_ACCOUNTS);
+
+    let (env, rpc_id) = setup(&accounts);
+    let TestLoopEnv { mut test_loop, datas: node_datas, tempdir } = env;
+
+    let mut nonce = 2;
+
+    // Deploy contracts for accounts at index 0, 10, 20, and so on.
+    let contract_ids =
+        accounts.iter().enumerate().filter(|(i, _)| i % 10 == 0).map(|(_, a)| a).collect_vec();
```

Comment:
nit: you can use iterator `step_by(10)` method

Reply:
simplified the test to make it more deterministic and scaled down the accounts.